### PR TITLE
Highlight the selected order in the order list

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #10595 

### Description

This PR highlights selected PR to visually distinguish the chosen order within the order list, specifically within a two-pane layout configuration. The highlighted order persists across pull-to-refresh, navigation across different tabs.

### Testing instructions

1. Open order listing screen in tablet device
2. Select any order item from the order list
3. Ensure the selected order is highlighted
4. Pull-To-Refresh and ensure the selected order item is still highlighted
5. Navigate across different tabs, come back to the order list and see the previously selected order item is still highlighted and it's corresponding screen is opened in detail pane

### Images/gif

[highlight_select_order.webm](https://github.com/woocommerce/woocommerce-android/assets/1331230/8e81ab8f-ea7e-4332-9297-a5eac6dac01c)

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>SelectedOrderTrackerViewModel</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by samiuelson — COMMENTED** [review]
❓ @AnirudhBhat, I think the colors are off in the dark theme. See the screenshot below:
Screenshot_20240207_151744

**Comment by AnirudhBhat** [comment]
>  I think the colors are off in the dark theme. See the screenshot below:

Good catch! I've fixed it in the following commits
https://github.com/woocommerce/woocommerce-android/pull/10643/commits/d5c409d54f0abe026472df80fcc321432a514eaa

https://github.com/woocommerce/woocommerce-android/pull/10643/commits/b8a8cbccff7212e7cf682acc4dca2e8422728a1d

https://github.com/woocommerce/woocommerce-android/pull/10643/commits/c6f51b9b9f4d93e811c377a0a04bdd9511361461

**Comment by AnirudhBhat** [comment]
@samiuelson Please ignore https://github.com/woocommerce/woocommerce-android/pull/10643/commits/1210e19a805c0b908c1ecb7a8d00605f61a1ec66 https://github.com/woocommerce/woocommerce-android/pull/10643/commits/17ac4b627305808222aec64d99691cfece1fe532 commits as it was pushed by mistake. I've reverted the changes.

File: WooCommerce/src/main/res/values/colors_base.xml [hunk 1]
```diff
@@ -45,6 +45,12 @@
     <color name="color_back_arrow">@color/color_on_surface</color>
     <color name="color_text_link">@color/woo_pink_50</color>
 
+    <!--
+        Color resource used to indicate selected state of an item
+    -->
+    <color name="woo_item_selected">@color/woo_purple_10</color>
+    <color name="color_item_selected">@color/woo_item_selected</color>
```
Comment (samiuelson) [inline WooCommerce/src/main/res/values/colors_base.xml:52]: ❓ Is it used anywhere?

**Review by samiuelson — APPROVED** [review]
Looks good overall. Left one question.

File: WooCommerce/src/main/res/values/colors_base.xml (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/res/values/colors_base.xml:52]: It should be `color_item_selected` in `OrderListAdapter`. Fixed it here: https://github.com/woocommerce/woocommerce-android/pull/10643/commits/12103ade3df9719a451cffff95ee3bd4a29f8b13

**Pull request merged**